apps/src/lib/client/utils.rs

```diff
@@ -263,11 +264,11 @@ pub async fn join_network(
     if let Some((validator_alias, pre_genesis_wallet)) =
         validator_alias_and_pre_genesis_wallet
     {
-        let tendermint_node_key: ed25519::SecretKey = pre_genesis_wallet
+        let tendermint_node_key: common::SecretKey = pre_genesis_wallet
             .tendermint_node_key
             .try_to_sk()
             .unwrap_or_else(|_err| {
-                eprintln!("Tendermint node key must be ed25519");
+                eprintln!("Tendermint node key must be common (need to change?)");
```

Comment:
ah, it might actually work with secp256k1, but we'll need to test that (we just cannot use the `common` signing scheme directly with tendermint as it has different encoding)

Reply:
Discussed yesterday with @tzemanovic : going to leave `tendermint_node_key` as a `common::SecretKey` and then instead change to a particular variant inside `fn id_from_pk(pk: &common::PublicKey)` before returning the `TenderminNodeId`

Reply:
Actually, why do we have here `tendermint_node_key` as a `SecretKey` but then we load it into `id_from_pk()` which takes a `PublicKey`? Are there foreseeable issues with this?

Reply:
Isn't `tendermint_node_key` as `SecretKey` because it is used as a parameter to `write_tendermint_node_key` on line 305 of `utils.rs`? I believe the `ref_to` (implemented in each signing scheme) call safely handles the conversion from `SecretKey` to `PublicKey`...

Reply:
Ah of course, I overlooked that. Thx